pkg/apis/pipeline/v1beta1/taskref_validation.go

```diff
@@ -54,6 +56,11 @@ func (ref *TaskRef) Validate(ctx context.Context) (errs *apis.FieldError) {
 	} else {
 		if ref.Name == "" {
 			errs = errs.Also(apis.ErrMissingField("name"))
+		} else {
+			// TaskRef name must be a valid k8s name
+			if errSlice := validation.IsQualifiedName(ref.Name); len(errSlice) != 0 {
```

Comment:
I'm quite surprised that golangci-lint allows such nested if then else to pass 😮 . 

It could be better if we can use switch case here the same way as we do in  `v1.taskref_validation.go()`?

Reply:
I think it's because it's an if/else but there was no else if. this logic is a bit different from v1 due to bundles but I refactored to use a switch statement.